CRM/Contact/Form/Task/SaveSearch.php

```diff
@@ -232,6 +236,20 @@ public function postProcess() {
 
     CRM_Contact_BAO_Group::create($params);
 
+    // Update mapping with the name and description of the group.
+    if ($mappingId) {
+      $mappingParams = array(
+        'id' => $mappingId,
+        'name' => CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_Group', $group->id, 'name', 'id'),
```

Comment:
Getting a fatal error here 
```
Oct 26 11:15:51  [info] $backTrace = #0 /Users/monish/src/civicrm/CRM/Core/Error.php(373): CRM_Core_Error::backtrace("backTrace", TRUE)
#1 /Users/monish/src/civicrm/CRM/Core/DAO.php(1062): CRM_Core_Error::fatal()
#2 /Users/monish/src/civicrm/CRM/Contact/Form/Task/SaveSearch.php(244): 
```
when I try to save a smart group on some Search Builder criteria because there is no such group object as ```$group``` here and I think we should fetch the value of ```description``` and ```name``` from submitted value (as ```$params```) respectively.